text/0000-roadmap-2017.md

```diff
@@ -0,0 +1,587 @@
+roadmap. That thread also posed a number of possible additional goals. Of
+course, part of the work of the roadmap is to allocate our limited resources,
+which fundamentally means not including some possible goals. Some of the most
+promising suggestions that didn't make it into the roadmap proposal itself are
+included in the Alternatives section.
+
+### Rust should have a lower learning curve
+
+Rust offers a unique value proposition in part because it offers a unique
+feature: its ownership model. Because the concept is not (yet!) a widespread one
+in other languages, it is something most people have to learn from scratch
+before hitting their stride with Rust. And that often comes on top of other
+aspects of Rust that may be less familiar. A common refrain is "the first couple
+of weeks are tough, but it's oh so worth it." How many people are bouncing off
+of Rust in those first couple of weeks? How many team leads are reluctant to
+introduce Rust because of the training needed? (1 in 4 survey respondents
+mentioned the learning curve.)
+
+Here are some strategies we might take to lower the learning curve:
+
+- **Improved docs**. While the existing Rust book has been successful, we've
+  learned a lot about teaching Rust, and there's a
+  [rewrite](http://words.steveklabnik.com/whats-new-with-the-rust-programming-language)
+  in the works. The effort is laser-focused on the key areas that trip people up
+  today (ownership, modules, strings, errors).
+
+- **Improved errors**. We've already made some
+  [big strides](https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html)
+  here, particularly for ownership-related errors, but there's surely more room
+  for improvement.
+
+- **Improved language features**. There are a couple of ways that the language
+  design itself can be oriented toward learnability. First, we can introduce new
+  features with an explicit eye toward
+  [how they will be taught](https://github.com/rust-lang/rfcs/pull/1636). Second,
+  we can improve existing features to make them easier to understand and use --
+  things like non-lexical lifetimes being a major example. There's already been
+  [some discussion on internals](https://internals.rust-lang.org/t/roadmap-2017-productivity-learning-curve-and-expressiveness/4097/)
+
+- **IDEs and other tooling**. IDEs provide a good opportunity for deeper
+  teaching. An IDE can visualize errors, for example *showing* you the lifetime
+  of a borrow. They can also provide deeper inspection of what's going on with
+  things like method dispatch, type inference, and so on.
+
+### Rust should have a pleasant edit-compile-debug cycle
+
+The edit-compile-debug cycle in Rust takes too long, and it's one of the
+complaints we hear most often from production users. We've laid down a good
+foundation with [MIR] (now turned on by default) and [incremental compilation]
+(which recently hit alpha). But we need to continue pushing hard to actually
```

Comment:
I think the idiomatic way to write this is `[incremental compilation][] (which recently hit alpha)`, for example:

[incremental compilation][] (which recently hit alpha).

[incremental compilation]: https://blog.rust-lang.org/2016/09/08/incremental.html